File: files/2024-09-15/CSEC 15.09.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge September 15th.xlsx"
input = read_excel(path, range = "B2:C9")
test  = read_excel(path, range = "E2:F6") %>% na.omit()

result = input %>%
  mutate(`Salary Range` = cut(x = Salary, 
                      breaks = c(-Inf, 1000, 5000, 10000, 15000, Inf), 
                      labels = c("<1000", "1000 - 4999", "5000 - 9999", "10000 - 14999", "> 15000"))) %>%
  summarise(Staffs = paste0(Staff, collapse = ", "), 
            .by = `Salary Range`)

cbind(result, test)
# identical, order of names in concatenation is different.

File: files/2024-09-15/CSEC 15.09.py
import pandas as pd

path = "files/Excel Challenge September 15th.xlsx"

input = pd.read_excel(path, usecols = "B:C", skiprows = 1, nrows = 7)
test = pd.read_excel(path, usecols = "E:F", skiprows = 1, nrows = 4).fillna("")

result = input.assign(Salary_Range = lambda x: pd.cut(x["Salary"], 
                                                      bins = [1000, 5000, 10000, 15000, 100000], 
                                                      labels = ["1000 - 4999", "5000 - 9999", "10000 - 14999", "> 15000"]))
result = result.groupby("Salary_Range")["Staff"].apply(lambda x: ", ".join(x)).reset_index(name="Staffs")

print(result)
print(test)
# sorting names in one cell is not the same.